### DISCUSSION TITLE
Monitor keeps deleting files and recreating files and folders

### DISCUSSION THREAD

**Discussion by mweiss219** [abraunegg/onedrive#2968] [General]
Hi
I just installed this OneDrive client on my Ubuntu server.
I need to synch one folder with OneDrive. I used the following command
```
onedrive --monitor --single-directory "/var/opt/mssql/data/backups/" --upload-only --verbose --enable-logging.
```

Checking the logs it keeps deleting and uploading again same files
```
2024-Nov-13 15:22:28.8121952	Sync with OneDrive is complete
2024-Nov-13 15:27:28.8459039	Starting a sync with OneDrive
2024-Nov-13 15:27:28.8459647	Syncing changes from this selected path: /var/opt/mssql/data/backups/
2024-Nov-13 15:27:28.8459872	Uploading differences of /var/opt/mssql/data/backups/
2024-Nov-13 15:27:28.8462654	Processing var/opt/mssql/data/backups
2024-Nov-13 15:27:28.8463031	The directory appears to have been deleted locally .. but we are running in --monitor mode. This may have been 'moved' on the local filesystem rather than being 'deleted'
2024-Nov-13 15:27:28.8463185	Deleting item from OneDrive: var/opt/mssql/data/backups
2024-Nov-13 15:27:29.4677961	Uploading new items of /var/opt/mssql/data/backups/
2024-Nov-13 15:27:29.4679191	OneDrive Client requested to create remote path: /var/opt/mssql/data/backups/
2024-Nov-13 15:27:29.6372247	The requested directory to create was not found on OneDrive - creating remote directory: /var/opt/mssql/data/backups/
2024-Nov-13 15:27:29.9014672	Successfully created the remote directory /var/opt/mssql/data/backups/ on OneDrive
2024-Nov-13 15:27:32.779082	Uploading new file…
```
Same happens every 5 minutes

Please advise 
Thanks

**Comment by abraunegg** [comment]
@mweiss219 
Lets start with some basic items:

1. What application version? What is the output of `onedrive --version` ?
2. Have you read all the documentation?

When you use this command: `--single-directory "/var/opt/mssql/data/backups/"` this path is *relative* to your 'sync_dir' path. This is *explained* in the documentation.

This means you are attempting to upload (technically): `~/OneDrive/var/opt/mssql/data/backups/` .....

**Comment by mweiss219** [comment]
Thanks for your quick response

1. Version is onedrive v2.4.13-1build1
2. Yes I've read the documentation, but I guess I missed something.

I taught that “/” at the beginning of the directory name will go to the system root.
What would be the correct syntax for an absolute path? 
It is actually uploading the correct files. but it's doing it recursively. 

Thanks again

**Comment by abraunegg** [comment]
@mweiss219 
> 1. Version is onedrive v2.4.13-1build1

This IS your problem

> 2. Yes I've read the documentation, but I guess I missed something.

Please re-read this *very* carefully: https://github.com/abraunegg/onedrive?tab=readme-ov-file#supported-application-version

Then install the application in the correct manner for your platform.

**Comment by abraunegg** [comment]
@mweiss219 
Once you correct your initial issues

> What would be the correct syntax for an absolute path?

There is no *absolute* path specification when using `--single-directory` it is all relative to 'sync_dir'

Your option is to configure 'sync_dir" corectly for your use case

**Comment by mweiss219** [comment]
I've just followed the install instructions for Ubuntu 22.04
https://github.com/abraunegg/onedrive/blob/master/docs/install.md

![image](https://github.com/user-attachments/assets/347f4369-af55-4cd1-9a2b-ddc29f1d9729)

Any instructions how to update?

Thanks

**Reply by abraunegg** [reply]
Please re-read ... especially that last cell on the right

**Comment by mweiss219** [comment]
This is exactly what I did. Anyway, I just updated my version now is onedrive v2.5.2-1+np1+1.1 

Thanks again and sorry for the bother.

If you would please advise the correct path for my scenario.
I need one-way synch upload only for this one directory /var/opt/mssql/data/backup/, and would run it as a background service via systemd

I guess I set the sync_dir to “/”
And add var/opt/mssql/data/backup to sync_list

Is that correct?

**Reply by abraunegg** [reply]
100% not correct. Do not use that approach.

What is your *desired* state online?

If you just want the files from that path in the 'root' of your online OneDrive display, configure 'sync_dir' to your path that you want to sync

If you want the files from that local path to be uploaded to a specific location online (recommendation) you have 2 options:

1. Leave sync_dir as default and create a symbolic link in ~/OneDrive to that folder
2. Create a folder (or path structure) in ~/OneDrive and as part of your backup process rsync the files from the backup path to this location

**Comment by mweiss219** [comment]
That really helped. Using the symbolic link option. seems to be working fine.
Thanks again
